src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -315,8 +310,12 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    task = _bufferedContent.CopyToAsync(stream).ContinueWithStandard(_bufferedContent, (copyTask, state) =>
```

Comment:
In the previous version that used ToArray, ToArray would copy out of the underlying buffer from beginning to end of the stream, regardless of the current position of the stream. This new version is only copying from the current position, and then it's setting the position to 0.  Is it guaranteed that upon entering this method that _bufferedContent.Position == 0?  If yes, since the logic here is based on it, there should be a debug asser to that effect.  If not, then this new version is wrong.